### ISSUE TITLE
Add to VS Code

### ISSUE THREAD

**Issue by jeff-hykin** [jeff-hykin/better-snippet-syntax#3]
@alexr00 This syntax might be something that VS Code wants to ship with.

I recently overhauled it to exaclty match the BNF spec provided by the VS Code docs.

**Comment by alexr00** [comment]
Thanks for the heads up! Is this intended to be for `.code-snippets` files or for all json with comments?

**Comment by jeff-hykin** [comment]
Actually that was the blocker wanted to discuss/request in this thread. Its definitely only intended for code snippets.

Maybe this has changed recently, but at least for me at the moment snippet files don't have any special extension (e.g. `javascript.json` instead of `javascript.code-snippets.json`). I was going to ask if such an extension could be added since, yes, I'm currently having to target all jsonc files as a workaround at the moment. It only highlights after it sees a "body" key, but I would still consider the false positives a blocker before adding it as a built-in.

TLDR; If there is a snippet extension, I'd like to target that.

**Comment by jeff-hykin** [comment]
Now that I look at it, the snippets are highlighted with jsonc even though their extension is .json. So I suppose there's something overriding the normal defaults, which I guess could also be used to avoid this issue.

**Comment by RedCMD** [comment]
the `json` and `jsonc` syntax highlighters are exactly the same, except `jsonc` has a `.comments` postfix added to (almost) all scope names
![image](https://github.com/jeff-hykin/better-snippet-syntax/assets/33529441/45d32f0e-77a1-4442-98c4-53879d4d48ab)

**Comment by alexr00** [comment]
I'm still trying to track down why snippet files are categorized as jsonc.

**Comment by alexr00** [comment]
Looks like it's coming from here: https://github.com/microsoft/vscode/blob/66b54e16972c2139980e26bc840940eca4a37862/extensions/configuration-editing/package.json#L59-L60

Based on that, we should be able to target those files + any that end in `.code-snippets` with a snippet grammar.

**Comment by alexr00** [comment]
I've started a draft PR to add this, but I'm going to get some more feedback from the dev team to see if we want to include this a language since it will always show in the language picker.

https://github.com/microsoft/vscode/pull/185017

**Comment by alexr00** [comment]
This is complete. The grammar will be in the August release (planned for next week).
